### ISSUE TITLE
Add line/col markers and disable line numbers by default

### ISSUE THREAD

**Issue by mhutch** [mono/monodevelop#4567]
We should add line/col numbers to the editor, and stop enabling line numbers by default. This would be more consistent with Visual Studio, would reduce visual noise in the editor, and would save valuable horizontal space. Additionally, the column indicator is a frequent request.

**Comment by vancura** [comment]
> I would reintroduce the status bar we already have it (for the closed pads). There is enough space at the left to add a line/column marker.

That's a more complex issue, and we should discuss that. But I am open to reintroducing footer.

> ATM we can't see the encoding for example.

I am not sure encoding is so important these days. @Therzok?

> An alternative would be to show it on top.

No, please. That's a very busy place already. Breadcrumbs, iOS Designer buttons. More to come.

/c @hbons @maryanne-exe

**Comment by mkrueger** [comment]
Encoding is more important these days since ever now we share "random encodings" from VS.NET

The problem with the path bar is that c#/VB/xml files have it other files don't. Like text or script files or unsupported languages.

**Comment by vancura** [comment]
@mkrueger asked me in our private Slack, so let's move it here:

> Why is the line black but the column is gray?

Column number is not as important as the line number. We're sort of decreasing importance to help clarify things and make the UI less crowded.

> this would make the path bar jump around

This is a serious point. As soon as the count of lines in the file changes above 9, 99, 999 (etc), the content right of the line number would shuffle. Also, it would shuffle when switching tabs.

**Comment by vancura** [comment]
We could potentially move the line/column counter in the footer which we already have there, to the right side (where we have zoom controls in Designers). 

That also means we should implement all these things together in one run: header, footer (with the view switcher), breadcrumbs (@chkn already wrote them, but they need some pixel tweaking, as they're a bit different from the original spec).

**Comment by mkrueger** [comment]
So 3 problems with the design:

1) The color choice looks weird - it's basically the coordinate having colors there doesn't make that much sense.
2) Modern editors show line, columns and selection size - the last one is pretty handy sometimes when trying to measure string lengths. I really miss that.
3) On the left side of the path bar is a very bad position because it gives that info too much presence - usually it's not needed. But the path is way more important than the cursor coodinates in 90% of the cases. Furthermore it would require to move the path to the right if the coodinates grow bigger which is IMO a bad design when witching between windows the position of the path bar should not move.

**Comment by mkrueger** [comment]
Depends we may show the path bar always as an option even for files which don't support it  - it would show 'no selection' in that case.

**Comment by vancura** [comment]
> Modern editors show line, columns, and selection size - the last one is pretty handy sometimes when trying to measure string lengths. I miss that.

Yes, that would be a useful feature – not only character size but maybe count of selected lines when in multi-line selection mode.

**Comment by vancura** [comment]
I'll draw how it would look in the footer, I just need to finish other stuff.

**Comment by vancura** [comment]
Another @mkrueger's good point: line number column on the left side of the source takes too much horizontal space, but _currently_ we do need it (we don't have counters). If we implement counters, we could set line column visibility to off by default as it's not really that important to show them at all times. Thoughts?

**Comment by mkrueger** [comment]
I would disable it as default.

However when line numbers are needed you really need line numbers - there is no way around it. (The same applies for the Column info we lost atm - but usually I count that manually like selection length :)).

The importance of line numbers 2018 is way less than 1998 - now the IDEs jump to things automatically and usually they're not needed. So it's a seldom used but very important feature. Like a drilling machine at home :)

**Comment by slluis** [comment]
We could show the line/column in a small overlay box at the bottom right corner of the editor. If the editor has a tab bar (like in the above screenshot), we could show it inside the bar, aligned to the right.

**Comment by mhutch** [comment]
For reference, here's a quick run-down of what VS Code has:

![image](https://user-images.githubusercontent.com/183285/39326835-33df9d94-4964-11e8-9028-e39e24677394.png)

One of the things I really like about their approach is that many of these are buttons that open up the command palette with some contextual actions.

* line number: jump to a line.
* encoding: reopen with different encoding, or convert encoding
* line endings: convert line endings
* language: switch language service used for file regardless of extension
* branch: change branch
* blame: a list of various git commands
* formatting: convert tabs to/from spaces, change tab size
* errors/warnings: open error pad
* status/notifications: show notifications

Obviously some of them wouldn't make sense in MD at all, such as the solution/configuration context, because we make that a top-level concept via the configuration selector and solution pad, and the notifications and build errors/warnings, because we put them in the status bar.

However, I think some of the others would be useful for sure. I've definitely had to resort to going outside the IDE to fix incorrectly detected file encoding, for example, or been frustrated by the lack of XML intellisense in an XML file without the .xml extension. So maybe it would be worth considering having a bar hosting several of these things.

**Comment by migueldeicaza** [comment]
The VSCode screenshot looks like an airplane cockpit, I rather not go down that path, so much of this is repetitive and conflicts with other designs (branch, errors, context, feedback).

Information about the document like Tab Size or UTF-8 BOM while practical, I feel like it belongs to the document properties, and we should put that there (On Apple, the universal command for getting there is Command-I).

I do like Lluis' suggestion of putting the line number/column on the bottom/right on the bar (I imagine that Lluis means is the area that on the screenshot has "Source", "Blame", etc).

**Comment by Therzok** [comment]
cc @abock

**Comment by vancura** [comment]
cc @sandyarmstrong
